### ISSUE TITLE
Stop worker letting it finish and submit current work.

### ISSUE THREAD

**Issue by nirvanaosc** [nicehash/excavator#29]
Hi,

I am making my own script to use Excavator through its web API.
I am looking for a way to stop a worker, but I want to let it finish and submit the current share.
The closest API method is "worker.free" but it doesn't seem to finish and submit current job.

Thank you.

**Comment by nicehashdev** [comment]
Hello!

Mining does not work as you are currently imaging. If share is found, it is immediatelly dispatched towards the pool. Miner does not do some continous work that takes time, it is only "guessing" by calculating hashes. It can make several hundred guesses per second (equihash) up to several billion guesses per second (decred, blake2s,...). If one guess is good, it is sent as a share.

**Comment by nirvanaosc** [comment]
Hi,

Thank you very much for your explanation. I contacted NiceHash requesting
this feature and they asked me to open an issue in the GitHub project.
I am a software engineer myself, but I don't know too much about how miners
work (I was thinking about making one myself but I found this one that is
quite good).
I know that the miner "guesses" calculagin hashes, but I thought that you
are given a piece of work and then you start calculating hashes until you
find one which difficulty is higgher than X, then you "solve" that piece of
work. Is that how it works? If so, aren't all unsuccessful tries like the
"progress" to solve that piece of work?

Thank you very much in advance.
